### ISSUE TITLE
Investigate options to set ArcGIS layers visibility to on

### ISSUE THREAD

**Issue by geographika** [compassinformatics/cpsi-mapview#592]
When accessing a layer such as https://gis.epa.ie/arcgis/rest/services/EPAMapServices/WaterFeatures/MapServer/ using the `BasiGX.view.form.AddArcGISRest` form, all sublayers are grouped together:

![image](https://user-images.githubusercontent.com/490840/184886543-f7a0fe4e-fb2a-4d1a-8334-e1dd73dac309.png)

This requests tiles in the format: https://gis.epa.ie/arcgis/rest/services/EPAMapServices/WaterFeatures/MapServer/export?dpi=96&transparent=true&format=png32&layers=show%3A2&bbox=-870963.0119157883%2C6911089.850464204%2C-814514.2665272616%2C6945448.294678345&bboxSR=102100&imageSR=102100&size=1477%2C899&f=image

However many sublayers have their visibility set to false by default and so cannot be displayed, e.g. https://gis.epa.ie/arcgis/rest/services/EPAMapServices/WaterFeatures/MapServer/4 has "Default Visibility: false".
These can be turned on using the following querystring: `layers=show:1,2,3`

For example the following shows the same tile with all layers turned on:

https://gis.epa.ie/arcgis/rest/services/EPAMapServices/WaterFeatures/MapServer/export?dpi=96&transparent=true&format=png32&layers=show:1,2,3,4,5,6,7&bbox=-870963.0119157883%2C6911089.850464204%2C-814514.2665272616%2C6945448.294678345&bboxSR=102100&imageSR=102100&size=1477%2C899&f=image

I'm unsure if there is a ``layers=show:all` option available that could be used. Otherwise individual layers would need to be displayed, or allow some way of turning all layers on by default.

**Comment by geographika** [comment]
@JakobMiksch - very nice! 

Yes I see how having those two buttons with similar labels would be confusing. Maybe change "Request available layers" to "List Map Services". 
"Add chosen layers" maybe change to "Add Selected Layers". 

> shall the sublayers only be visible when the layer is checked?

Not sure if this refers to the layer tree on the left, or the layers in the "Available layers" section of the form?

**Comment by JakobMiksch** [comment]
> > shall the sublayers only be visible when the layer is checked?
> 
> Not sure if this refers to the layer tree on the left, or the layers in the "Available layers" section of the form?

I was referring to the "Available layer" section in the form. My though is that it only makes sense to edit the sublayer's visibilities if the main layer is checked.

**Comment by geographika** [comment]
> > > shall the sublayers only be visible when the layer is checked?
> > 
> > 
> > Not sure if this refers to the layer tree on the left, or the layers in the "Available layers" section of the form?
> 
> I was referring to the "Available layer" section in the form. My though is that it only makes sense to edit the sublayer's visibilities if the main layer is checked.

Yes that makes sense to me. If the top level isn't selected then it shouldn't be able to add it in to the map anyway.

**Comment by JakobMiksch** [comment]
@geographika The current state of the development is in this PR https://github.com/compassinformatics/cpsi-mapview/pull/626
It an be tested already.

**Comment by JakobMiksch** [comment]
This new arrangement of the buttons makes sense to me. Before it often happened to me that I clicked the wrong button.

**Comment by geographika** [comment]
@fergaldoyle - yes, that approach seems much more intuitive. The original design was based on the WMS loader `BasiGX.view.form.AddWms` (see below). Maybe we can suggest this change to that loader too in the future. 

![image](https://user-images.githubusercontent.com/490840/216548303-c8dd5d37-7d0f-4337-a71c-1cb8a8c1970d.png)

**Comment by fergaldoyle** [comment]
@geographika @JakobMiksch 

When adding ARCGIS external layers, the item(s) added to the layer selector is the service name, e.g. "Copernicus/HotSpotMonitoring", and gives no insight into what layers inside the service were selected / visible.

e.g. In the screenshot below I've selected
Copernicus/HighResolutionLayers > Grassland
Copernicus/HighResolutionLayers > Forest
Copernicus/HotSpotMonitoring > Land Cover - Dublin
![selected-layers-1](https://user-images.githubusercontent.com/1889044/217608016-7897a54c-9048-41ea-b49c-1af6c34806ec.png)

But the layer selector only shows the service name.  Might it be an idea to have the service name as the category / folder, then selected layers shown within each category:
![selected-layers-2](https://user-images.githubusercontent.com/1889044/217609207-fcfa110b-4de7-43fc-ae28-4abbb30d9fe4.png)

In the case of https://gis.epa.ie/arcgis/rest/services/Copernicus/HotSpotMonitoring/MapServer, there's layers and sublayers. 
The default ArcGIS viewer handles these levels of nesting like this:
![arc-layer-list](https://user-images.githubusercontent.com/1889044/217610082-6d22d78d-7f60-44eb-b6fb-f18d663259f5.png)

Maybe the Add ARCGIS Rest Layer form could be simplified. If you select a service, e.g. Copernicus/HighResolutionLayers, then a category/folder is added to the External Layers category with name Copernicus/HighResolutionLayers, and the layers and sublayers hierarchy within Copernicus/HighResolutionLayers service are added to the Copernicus/HighResolutionLayers folder under External Layers, similar to the ArcGIS viewer. The default checked state of each layer is taken from "Default Visibility" property of each layer.  This removes the need for the checkbox, sub-checkbox interface in the ARCGIS Rest layer form, and also  retains the hierarchy which is present ArcGIS. It also removes some unhandled cases like where a user adds layers from a service, then comes back to add more from the same service,
